### PR TITLE
Fix to comply with new endpoints (WithSites, etc)

### PR DESCRIPTION
**Pull request by AlixH**

### CODE REVIEW

File: src/provider/CentralServerProvider.tsx [hunk 1]
```diff
@@ -514,15 +514,13 @@ export default class CentralServerProvider {
     return result.data;
   }
 
-  public async getChargingStation(id: string): Promise<ChargingStation> {
+  public async getChargingStation(params: Record<string, any>): Promise<ChargingStation> {
```
Comment (LucasBrazi06) [inline src/provider/CentralServerProvider.tsx:517]: The ID should stay in the method's signature and you can add an extra param as optional with the type used in the backend (HttpChargingStationRequest).

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

File: src/provider/CentralServerProvider.tsx (hunk 1 above)
Comment (AlixH) [reply src/provider/CentralServerProvider.tsx:517]: I don't understand, this type only has an ID field

Comment (LucasBrazi06) [reply src/provider/CentralServerProvider.tsx:517]: Yes, you're right, you can use the ev-server's type in ev-mobile and remove the ID.
I like to see the mandatory params visible in the method's signature (the ID here.)

@Ninjeneer, we should also do the same in Angular central-server.service class.
So the types are aligned with the backend:
`getChargingStation(id: string, params?: HttpChargingStationRequest) {...}`
What do you think?

Comment (AlixH) [reply src/provider/CentralServerProvider.tsx:517]: I find it a bit confusing to call the second param 'params', as the ID is also part of the HTTP request params

Comment (LucasBrazi06) [reply src/provider/CentralServerProvider.tsx:517]: > I find it a bit confusing to call the second param 'params', as the ID is also part of the HTTP request params

This is what I explain above, remove the ID in the front-end type (the backend has all but not the fronts)
We can have a call?

Comment (AlixH) [reply src/provider/CentralServerProvider.tsx:517]: Yes for a call

Comment (AlixH) [reply src/provider/CentralServerProvider.tsx:517]: As discussed with Serge, let's use getChargingStation(id: string, extraParams?: HttpChargingStationRequest) {...}

File: src/provider/CentralServerProvider.tsx [hunk 2]
```diff
@@ -514,15 +510,14 @@ export default class CentralServerProvider {
     return result.data;
   }
 
-  public async getChargingStation(id: string): Promise<ChargingStation> {
+  public async getChargingStation(ID: string, extraParams: HttpChargingStationRequest = {}): Promise<ChargingStation> {
```
Comment (LucasBrazi06) [inline src/provider/CentralServerProvider.tsx:513]: Rename ID -> id ;-)

File: src/screens/transactions/chart/TransactionChart.tsx [hunk 3]
```diff
@@ -115,8 +115,11 @@ export default class TransactionChart extends BaseAutoRefreshScreen<Props, State
 
   public getChargingStation = async (chargingStationID: string): Promise<ChargingStation> => {
     try {
+      const params = {
```
Comment (LucasBrazi06) [inline src/screens/transactions/chart/TransactionChart.tsx:118]: Rename params -> extraParams (your idea btw)

File: src/types/ChargingStation.tsx [hunk 4]
```diff
@@ -140,3 +140,8 @@ export enum ConnectorCurrentLimitSource {
   STATIC_LIMITATION = 'SL',
   CONNECTOR = 'CO'
 }
+
+export interface HttpChargingStationRequest {
```
Comment (LucasBrazi06) [inline src/types/ChargingStation.tsx:144]: Like in the backend, create a folder named requests and put all Http types in it.

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

File: src/provider/CentralServerProvider.tsx [hunk 5]
```diff
@@ -514,15 +511,14 @@ export default class CentralServerProvider {
     return result.data;
   }
 
-  public async getChargingStation(id: string): Promise<ChargingStation> {
+  public async getChargingStation(id: string, extraParams: HttpChargingStationRequest = {}): Promise<ChargingStation> {
     this.debugMethod('getChargingStation');
-    const url = this.buildRestEndpointUrl(ServerRoute.REST_CHARGING_STATION, { id });
+    const params = { ID: id, ...extraParams };
```
Comment (LucasBrazi06) [inline src/provider/CentralServerProvider.tsx:516]: Je chipote: no need to create an intermediary var here, you can just add one line to the previous existing code (more readable)

```
    const result = await this.axiosInstance.get(url, {
      headers: this.buildSecuredHeaders(),
      params: {
        ID: id
        ...extraParams
      }
```

File: src/screens/transactions/chart/TransactionChart.tsx [hunk 6]
```diff
@@ -115,8 +115,11 @@ export default class TransactionChart extends BaseAutoRefreshScreen<Props, State
 
   public getChargingStation = async (chargingStationID: string): Promise<ChargingStation> => {
     try {
+      const extraParams = {
```
Comment (LucasBrazi06) [inline src/screens/transactions/chart/TransactionChart.tsx:118]: Should be typed HttpChargingStationRequest

`const extraParams: HttpChargingStationRequest = {...`

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
Almost done ;-)

**Review by LucasBrazi06 — APPROVED** [review]

**Pull request merged**